### ISSUE TITLE
Strahler order fails on a simplified neuron with no branches

### ISSUE THREAD

**Issue by dokato** [natverse/nat#464]
For a neuron with a single branch (e.g. attached) I'd expect Strahler's order to return 1 per each node. Instead it errors out:

```r
> nn
'neuron' with 403 vertices in 1 tree
> plot3d(nn,soma=T)
> summary(nn)
  root nodes segments branchpoints endpoints cable.length
1    1   403        1            0         2     206.8101
> strahler_order(nn)
Error in idxs[i, ] : incorrect number of dimensions
```

[single_branch_neuron.rds.zip](https://github.com/natverse/nat/files/6575898/single_branch_neuron.rds.zip)

**Comment by dokato** [comment]
idea for a quick win:
```r
  if (summary(branchpoints)$branchpoints == 0)
    return(rep(1, summary(nn)$nodes))
```

at the beginning of `strahler_order` function.

**Comment by jefferis** [comment]
`summary(x)$branchpoints` is actually sort of slow because it does some other stuff. 

```
branchpoints=length(object$BranchPoints)
```
is the relevant bit, so you could use that. I forget how it handles neurons with multiple trees though.

**Comment by jefferis** [comment]
Closed by #465
